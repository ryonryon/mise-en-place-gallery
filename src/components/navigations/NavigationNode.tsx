import styled from "@emotion/styled";
import * as React from "react";

interface Props {
  item: string;
  childItems: string[] | null;
  nodeLevel: number;
  display: boolean;
}

const NavigationNode = ({ item, childItems, nodeLevel, display }: Props) => {
  const [childNodesVisiblity, setChildNodesVisiblity] = React.useState(false);

  const NavigationNodeDiv = styled.div`
    padding: 16px 0 16px ${16 + nodeLevel * 8}px;
    ${1 < nodeLevel ? "color: #858585;" : ""}
    ${display ? "" : "display: none;"}
  `;
  return (
    <>
      <a href={`#${item.toLowerCase()}`}>
        <NavigationNodeDiv
          onClick={() => {
            setChildNodesVisiblity(prev => (prev ? false : true));
          }}
        >
          {item}
        </NavigationNodeDiv>
      </a>
      {childItems
        ? childItems.map((child, index) => (
            <NavigationNode key={index} item={child} nodeLevel={nodeLevel + 1} display={childNodesVisiblity} />
          ))
        : null}
    </>
  );
};

NavigationNode.defaultProps = {
  childItems: null,
  nodeLevel: 1,
  display: true
};

export default NavigationNode;
