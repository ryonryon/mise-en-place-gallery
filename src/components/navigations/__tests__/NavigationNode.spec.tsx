import * as React from "react";
import * as renderer from "react-test-renderer";

import NavigationNode from "../NavigationNode";

describe("NavigationNode test", () => {
  it("tests NavigationNode render correctly", () => {
    const navigationNode = renderer.create(<NavigationNode item="item" />);

    expect(navigationNode).toMatchSnapshot();
  });
});
