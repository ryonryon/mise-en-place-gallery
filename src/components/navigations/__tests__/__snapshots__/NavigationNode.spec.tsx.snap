// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationNode test tests NavigationNode render correctly 1`] = `
<a
  href="#item"
>
  <div
    className="css-72aj2j"
    onClick={[Function]}
  >
    item
  </div>
</a>
`;
